pkg/controller/operators/olm/operator.go

```diff
@@ -2087,7 +2098,15 @@ func (a *Operator) transitionCSVState(in v1alpha1.ClusterServiceVersion) (out *v
 		}
 
 		// If there is a succeeded replacement, mark this for deletion
-		if next := a.isBeingReplaced(out, a.csvSet(out.GetNamespace(), v1alpha1.CSVPhaseAny)); next != nil {
+		next := a.isBeingReplaced(out, a.csvSet(out.GetNamespace(), v1alpha1.CSVPhaseAny))
+		if operatorGroup.UpgradeStrategy() == operatorsv1.UnsafeFailForwardUpgradeStrategy {
```

Comment:
Might be worth segregating this to its own method `IsFailForwardEnabled` or some such. Just to make it easy to refactor later in case something changes =P

Reply:
This is what we do with the downstream package server manager component that's solely responsible for reading the topology mode from this cluster-wide API and returning a bool for whether single node functionality should be enabled. Seems relevant here but not blocking imo.